### PR TITLE
Add current CLI commands to `packages/cli/README.md`

### LINKED ISSUE #301: Add current CLI commands to `packages/cli/README.md`
The doc does not include the commands (with options) available along with a brief description for each. See: 
https://github.com/redwoodjs/redwood/blob/master/packages/cli/README.md

- Also needs to include info that we use [Yargs](https://github.com/yargs/yargs) for the CLI.
- Should mention `—help` flag.

Running `yarn rw —help` outputs this list:
```
Commands:
  rw build [app..]    Build for production.
  rw db <command>     Database tools.                        [aliases: database]
  rw dev [app..]      Run development servers.
  rw generate <type>  Save time by generating boilerplate code.     [aliases: g]
  rw lint             Lint your files.
  rw open [port]      Open your project in your browser.
  rw test [app..]     Run Jest tests for api and web.
```

And here’s the cli/ package directory with all commands:
https://github.com/redwoodjs/redwood/tree/master/packages/cli/src/commands

### PR DESCRIPTION
**Pull request by AryanJ-NYC**
This pull request is a work in progress. I will mark it ready to review as soon as I can.

I opened this pull request to get comments on the documentation I have written so far. Great documentation requires some iteration so I thought I'd start that process as early as possible.

**List of CLI Commands**
- [x] redwood build
- [x] redwood db
  - [x] redwood db down
  - [x] redwood db generate
  - [x] redwood db save
  - [x] redwood db seed
  - [x] redwood db up
- [x] redwood dev
- [ ] redwood generate
  - [ ] redwood generate cell
  - [ ] redwood generate component
  - [ ] redwood generate layout
  - [ ] redwood generate page
  - [ ] redwood generate scaffold
  - [ ] redwood generate sdl
  - [ ] redwood generate service
- [x] redwood lint
- [x] redwood open
- [x] redwood test

I am yet to get around to those commands without the check.

Closes #301

### CODE REVIEW

**Comment by AryanJ-NYC** [comment]
Question: we have CLI development instructions [here](https://github.com/redwoodjs/redwood/blob/master/CONTRIBUTING.md#cli-package-development) and [here](https://github.com/redwoodjs/redwood/blob/master/packages/cli/README.md#development). We probably want one source of truth. Should I consolidate? If so, to `packages/CLI/README` or `/CONTRIBUTING.md`? There are valid arguments for both.

**Comment by AryanJ-NYC** [comment]
Question: The [CLI README](https://github.com/redwoodjs/redwood/blob/master/packages/cli/README.md#publishing) reads:
> This is a monorepo and is published via LernaJS. See the root README for instructions.

The root README does not have these instructions. Are they available anywhere? Should I just delete this section?

File: packages/cli/README.md [hunk 1]
```diff
@@ -1,19 +1,44 @@
...
 
-The Redwood CLI comes with RedwoodJS (which means no extra software to install!).
+The Redwood CLI comes with RedwoodJS (which means no extra software to install!). It is made with [yargs](https://yargs.js.org/).
 
 ## Usage
 
 The [`yarn`](https://classic.yarnpkg.com/en/docs/install) package is required to use the Redwood CLI.
 
-Be sure to prefix all Redwood CLI commands with `yarn `. For example, `yarn redwood new`.
+Be sure to prefix all Redwood CLI commands with `yarn`. For example, `yarn redwood new`.
```
Comment (peterp) [inline packages/cli/README.md]: `yarn rw new` doesn't exist anymore. We should probably remove it.

**Comment by peterp** [comment]
Thanks for this.

Your comment about the source of truth made me consider if adding documentation for each command (beyond what we already supply in the description) is valuable? @thedavidprice 

I can't imagine maintaining this in the README as well.

I think the CLI readme is useful for people who are trying to understand the purpose of this package.

File: packages/cli/README.md (hunk 1 above)
Comment (AryanJ-NYC) [reply packages/cli/README.md]: Done.

**Comment by AryanJ-NYC** [comment]
@peterp Great. In writing these docs, I kind of agree.

Perhaps the source of truth for how to use the Redwood CLI should be, well, the Redwood CLI. `yarn redwood --help` should be verbose enough to be documentation in and of itself (in my opinion).

With that thought, it may also help CLI users to implement some [completion shortcuts for commands and options](https://github.com/yargs/yargs/blob/master/docs/api.md#completioncmd-description-fn). This indirectly helps with documenting how the CLI is used and will probably be a big gain for developers.

With respect to this current PR, if we think the maintenance of this CLI README is a nightmare and is not very valuable, please don't be afraid to close it. There'll be no hurt feelings. I think a lot of good came out of it already.

**Comment by peterp** [comment]
@AryanJ-NYC Ok, and thanks! Like you said I don't think this is in vain since you're probably know a bunch about the CLI.

I think a completion shortcut would be amazing! Is there a way that completions can be generated from the commands that they import?

I opened up #311 in response to this.

**Comment by peterp** [comment]
I'm going to include the `[port]` fix, thanks for this!

**Comment by thedavidprice** [comment]
Starting up #322 to continue this conversation

**Comment by jtoar** [comment]
@AryanJ-NYC are you interested in picking this up again? I'm about to take a crack at CLI docs and wanted to loop you in before I started.

**Comment by AryanJ-NYC** [comment]
@jtoar Nope. Go for it!

**Pull request closed without merging**